### ISSUE TITLE
About "training_progress_scores.csv"

### ISSUE THREAD

**Issue by khttknoob** [ThilinaRajapakse/simpletransformers#400]
Can't seem to find this file in the output folder I want to get the accuracy,loss on per epoch and read somewhere I can find it in this file but can't seem to find it. Any kind of help will be appreciated.

It's my first time on github. So please do tell me if this is the wrong section.
Thank you :)

**Comment by ThilinaRajapakse** [comment]
There should be a file with the name `training_progress_scores` in your output directory.

You are in the right place! And welcome to GitHub!

**Comment by khttknoob** [comment]
@ThilinaRajapakse No its not there I cant seem to find it despite running the model a few times over and over again. So, I thought maybe there is a argument for it but still nothing.

**Comment by ThilinaRajapakse** [comment]
Which model are you using?

Can you share a minimal script?

**Comment by khttknoob** [comment]
`custom_args = {'fp16': False, # not using mixed precision 
               'train_batch_size': 8, # default is 8
               'gradient_accumulation_steps': 2,
               'do_lower_case': True,
               'save_model_every_epoch': False,
               'evaluate_during_training': False,
               'learning_rate': 1e-05, # using lower learning rate
               'overwrite_output_dir': True, # important for CV
               'num_train_epochs': 2} # default is 1

model = ClassificationModel('bert', 'bert-base-cased', num_labels=3, args=custom_args) 
    model.train_model(train_df)
    result, model_outputs, wrong_predictions = model.eval_model(val_df, acc=sklearn.metrics.accuracy_score)
    print(result['acc'])`

**Comment by khttknoob** [comment]
`custom_args = {'fp16': False, # not using mixed precision 
               'train_batch_size': 8, # default is 8
               'gradient_accumulation_steps': 2,
               'do_lower_case': True,
               'save_model_every_epoch': False,
               'evaluate_during_training': False,
               'learning_rate': 1e-05, # using lower learning rate
               'overwrite_output_dir': True, # important for CV
               'num_train_epochs': 2} # default is 1`

`model = ClassificationModel('bert', 'bert-base-cased', num_labels=3, args=custom_args) 
    model.train_model(train_df)
    result, model_outputs, wrong_predictions = model.eval_model(val_df, acc=sklearn.metrics.accuracy_score)
    print(result['acc'])`

**Comment by khttknoob** [comment]
Is this fine or should I upload it in a text file for you @ThilinaRajapakse  ?

**Comment by ThilinaRajapakse** [comment]
My bad for not realizing this straight away, but the `training_progress_scores.csv` file is only created when `evaluate_during_training` is used. I'll look into creating this file in all cases. I think there was an earlier issue asking about this as well, now that I think about it.

In the meantime, the easiest way to observe the training progress is to use visualization support with [W&B](https://www.wandb.com/).
https://github.com/ThilinaRajapakse/simpletransformers#visualization-support

The snippet you shared is fine. Just a tip, you can surround a code block with 3 backticks to make it appear as a code block. You can get syntax highlighting for different languages with GitHub markdown. (Edit your earlier comment to see how it's done).

**Comment by khttknoob** [comment]
@ThilinaRajapakse Thank you so much will try this and will let you know if I encounter any further problems with it :) . 

And can you check my snippet now is it fine ?

**Comment by ThilinaRajapakse** [comment]
Yeah, looks good!

**Comment by khttknoob** [comment]
@ThilinaRajapakse Getting this error 

**UserWarning: wandb_project specified but wandb is not available. Wandb disabled.
  warnings.warn("wandb_project specified but wandb is not available. Wandb disabled.")**

Have installed wandb and logged in too.

This is my code.
```pyton
custom_args = {'fp16': False, # not using mixed precision 
               'train_batch_size': 8, # default is 8
               'gradient_accumulation_steps': 2,
               'do_lower_case': True,
               'save_model_every_epoch': False,
               'evaluate_during_training': False,
               'wandb_project': "test",
               'learning_rate': 1e-05, # using lower learning rate
               'overwrite_output_dir': True, # important for CV
               'num_train_epochs': 1,} # default is 1```

**Comment by khttknoob** [comment]
@ThilinaRajapakse waiting for your reply Sir. Really need your help.

**Comment by ThilinaRajapakse** [comment]
It may be an issue with your environment. Wandb is not being detected. Are you using conda?

**Comment by khttknoob** [comment]
No Sir. I am using Google Colab.

**Comment by khttknoob** [comment]
@ThilinaRajapakse I tried everything. Can't seem to make it work :(

**Comment by ThilinaRajapakse** [comment]
Does a simple `import wandb` work?

**Comment by khttknoob** [comment]
@ThilinaRajapakse Yes, import wandb works.

**Comment by ThilinaRajapakse** [comment]
https://colab.research.google.com/drive/1RlsZ4ILHoUs3KYxi3eZROyVzr5p0zib9?usp=sharing

It works without an issue for me when using your own args.

**Comment by khttknoob** [comment]
Oh issue solved with creating a fresh notebook. But there is still the an issue beacuse Im not getting accuracy in the graph only getting global_step, Training loss and lr. Am I doing something wrong @ThilinaRajapakse ?

**Comment by ThilinaRajapakse** [comment]
https://simpletransformers.ai/docs/usage/#additional-evaluation-metrics

**Comment by khttknoob** [comment]
@ThilinaRajapakse Yes I am already using `sklearn.metrics.accuracy_score` to get the accuracy but how do I log it so it can show me a graph of epochs and there respective accuracy in W&B.
